pkg/apis/core/types.go

```diff
@@ -4142,11 +4142,48 @@ type NodeAddressType string
 
 // These are valid values of node address type
 const (
-	NodeHostName    NodeAddressType = "Hostname"
-	NodeExternalIP  NodeAddressType = "ExternalIP"
-	NodeInternalIP  NodeAddressType = "InternalIP"
-	NodeExternalDNS NodeAddressType = "ExternalDNS"
+	// NodeHostName identifies the node's canonical hostname
+	NodeHostName NodeAddressType = "Hostname"
+
+	// NodeInternalIP identifies an IP address which is assigned to one of the node's
+	// network interfaces. Every node should have at least one address of this type.
+	//
+	// An internal IP is normally expected to be reachable from every other node, but
+	// may or may not be visible to hosts outside the cluster. By default it is
+	// assumed that kube-apiserver can reach node internal IPs, though it is possible
+	// to configure clusters where this is not the case.
+	//
+	// In cloud environments, a NodeInternalIP address generally corresponds to an
+	// "internal" or "private" IP address which was automatically assigned to the
+	// node. In bare-metal clusters, NodeInternalIP is simply the standard address
```

Comment:
Is it worth documenting that the address associated with the default route is usually preferred?

Reply:
If you're only going to have _one_ node address then it should be the one associated with the default route, but that's not necessarily true about InternalIPs in general... (Like, if we were going to document that anywhere, it might make more sense to document it on the `NodeStatus.Addresses` field, not on `NodeAddressType`?)